precompiles/staking/method_unstake.go

```diff
@@ -0,0 +1,77 @@
+package staking
+
+import (
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	ptypes "github.com/zeta-chain/node/precompiles/types"
+)
+
+func (c *Contract) Unstake(
+	ctx sdk.Context,
+	evm *vm.EVM,
+	contract *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != 3 {
+		return nil, &(ptypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 3,
+		})
+	}
+
+	stakerAddress, ok := args[0].(common.Address)
+	if !ok {
+		return nil, ptypes.ErrInvalidArgument{
+			Got: args[0],
+		}
+	}
+
+	if contract.CallerAddress != stakerAddress {
+		return nil, fmt.Errorf("caller is not staker address")
+	}
+
+	validatorAddress, ok := args[1].(string)
+	if !ok {
+		return nil, ptypes.ErrInvalidArgument{
+			Got: args[1],
+		}
+	}
+
+	amount, ok := args[2].(*big.Int)
+	if !ok {
+		return nil, ptypes.ErrInvalidArgument{
+			Got: args[2],
+		}
+	}
+
+	msgServer := stakingkeeper.NewMsgServerImpl(&c.stakingKeeper)
+	res, err := msgServer.Undelegate(ctx, &stakingtypes.MsgUndelegate{
+		DelegatorAddress: sdk.AccAddress(stakerAddress.Bytes()).String(),
+		ValidatorAddress: validatorAddress,
+		Amount: sdk.Coin{
+			Denom:  c.stakingKeeper.BondDenom(ctx),
+			Amount: math.NewIntFromBigInt(amount),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	stateDB := evm.StateDB.(ptypes.ExtStateDB)
+	err = c.addUnstakeLog(ctx, stateDB, stakerAddress, validatorAddress, amount)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
What is the expected behaviour if a user , executes a `CancelUnbondingDelegation` after this . Within the unbending period